src/asciidoc/docs/develop/testingTemplates.adoc

```diff
@@ -0,0 +1,164 @@
+= Testing templates
+:page-slug: testing-templates
+:page-description: A consolidated resource for methods a developer can use to test templates
+
+There are many ways you can test your Koji templates.
+Typically, you will begin testing in the Koji editor.
+Later, you'll want to publish the template as *Unlisted*, so you can test it in a production environment before giving access to general users.
+Some of the most common testing methods are listed below.
+
+== Testing in the Koji editor
+
+The Koji editor provides various ways to preview a template's behavior.
+These are covered in more detail in the documentation for the editor (see <<editor#>>).
+
+=== Previewing the Koji
+
+The *Live preview* tab allows you to toggle between *Preview* mode, which shows what the Koji looks like when running, and *Editing* mode, which allows you to perform remixing in the preview.
+
+The *Remote* tab offers several additional ways of previewing your Koji.
+You can test it on your mobile device by scanning a QR code.
+There are also options for opening the Koji in a new tab, running it in a live player, or copying the link for the preview.
+If you share this link with someone else, they can preview the Koji while it's still in development.
+
+[NOTE]
+The remote preview link will only work for as long as your editor session is still active, so it is not suitable for long-term sharing.
+
+The *Routes* tab lets you test your backend routes.
+It's similar to using a program like curl or Fiddler.
+For instance, you can manually enter an HTTP request, configure a header for it, send the request, and examine the results returned by the server.
+
+=== Previewing different contexts
+
+A Koji can display different landing pages depending on the context in which it is running.
+For more on contexts, see <<withkoji-vcc-package#_feedsdk#>>.
+
+When running a Koji outside the editor, you can display its landing page by appending a query string to the URL.
+For instance, if your Koji has an admin page, you can display it by appending `?context=admin` to the URL.
+
+Inside the editor, the *Live preview* tab will display buttons for the following contexts if they exist in your Koji: *Admin*, *About*, *Sticker*, *Buyer Receipt*, *Seller Receipt*.
+Clicking the button for one of these contexts displays a preview of its landing page.
+
+You can use other names for custom contexts, but *Live preview* will not display buttons for them.
```

Comment:
This information is very helpful. The phrasing can be streamlined to make it clearer what the developer can do. Something like...
"The editor provides built-in previews for standard contexts. To preview a custom context, you can append the query string..."